spec/rubocop/cop/rails/enum_hash_spec.rb

```diff
@@ -1,162 +1,314 @@
 # frozen_string_literal: true
 
 RSpec.describe RuboCop::Cop::Rails::EnumHash, :config do
-  context 'when array syntax is used' do
-    context 'with %i[] syntax' do
-      it 'registers an offense' do
+  context 'when Rails 7 syntax is used' do
```

Comment:
I think there are generally no pragmatic issues, but it would be best if this new syntax is only detected in Rails 7.0 and later. In other words, it will not be detected in Rails 6.1 or earlier.